core/client/src/main/java/alluxio/client/lineage/AbstractLineageClient.java

```diff
@@ -77,11 +86,16 @@ public long createLineage(List<AlluxioURI> inputFiles, List<AlluxioURI> outputFi
   @Override
   public boolean deleteLineage(long lineageId, DeleteLineageOptions options)
       throws IOException, LineageDoesNotExistException, LineageDeletionException, AlluxioException {
-    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    LineageMasterClient masterClient = null;
     try {
+      masterClient = mContext.acquireMasterClient();
       boolean result = masterClient.deleteLineage(lineageId, options.isCascade());
       LOG.info("{} delete lineage {}", result ? "Succeeded to " : "Failed to ", lineageId);
       return result;
+    } catch (NotFoundException e) {
+      throw new LineageDoesNotExistException(e.getMessage());
+    } catch (AlluxioStatusException e) {
+      throw new LineageDeletionException(e.getMessage());
```

Comment:
changed to convert these a little better, probably not very important since I doubt anyone in the world is handling these exceptions differently.